modules/util.py

```diff
@@ -370,6 +370,22 @@ def assert_json_value(self, json_data, jsonpath_expr, expected_value):
             f"Expected {expected_value}, but got {match[0].value}",
         )
 
+    def trim_url_to_last_slash(self, url: str):
```

Comment:
I could be wrong, but I think the general case for what this method does is _actually_ "get the domain of a web resource" rather than "get the parent URL directory". Consider this instead:

```python
from urllib.parse import urlparse, urlunparse

def get_domain_from_url(self, url: str) -> str:
    parsed_url = urlparse(url)
    domain_parsed_url = parsed_url._replace(path="")
    return urlunparse(domain_parsed_url)
```

Up to you, but I think this is going to be more relevant

Reply:
oh yeah, thanks for the suggestion! I just implemented that